e2e/common/components/base-page.ts

```diff
@@ -1,63 +1,151 @@
+
+        return new promise.Promise((resolve, reject) => {
+            this.getDefaultConf().forEach( item => {
+                if (item.checkOnPropertyValue) {
+                    item.element.getAttribute('value').then( inText => {
+                        if (inText !== item.expectedValue) {
+                            toReturn = false;
+                            message += item.messageOnKO;
+                        }
+                    });
+                } else {
+                    item.element.getText().then( inText => {
+                        if (inText !== item.expectedValue) {
+                            toReturn = false;
+                            message += item.messageOnKO;
+                        }
+                    });
+                }
+            });
+
+            protractor.promise.controlFlow().execute( () => {
+                if (toReturn !== true) {
+                    reject(message);
+                } else {
+                    resolve(true);
+                }
+            });
+        });
+    }
+    public getAllButtons(): ElementArrayFinder {
+        return new ElementArrayFinder(browser);
+    }
+
+    public getMainWindow() {
+        return element.all(by.tagName(this.tagName))
+            .get(this.tagNameIndex);
+    }
+
+    public getTitle() {
+        return this.getMainWindow()
+            .element(by.tagName('systelab-dialog-header'))
+            .element(by.className('slab-dialog-header'));
+    }
+
+    public getButtonClose() {
+        return this.getMainWindow()
+            .element(by.className('slab-dialog-close'));
+    }
+
+    public getObjectById(id: string) {
+        return this.getMainWindow()
+            .element(by.id(id));
+    }
+
+    public getObjectByDataTestId(customId: string) {
+        return element.all(by.css('[data-test-id="' + customId + '"]'));
+    }
+
+    public getButtonByName(name: string) {
+        return this.getMainWindow()
+            .element(by.buttonText(name));
+    }
+
+    public checkPresentAndDisplayed() {
+        expect(this.getMainWindow()
+            .isPresent())
+            .toEqual(true, 'Window should be present on the DOM');
+        expect(this.getMainWindow()
+            .isDisplayed())
+            .toEqual(true, 'Window should be displayed');
+    }
+
+    public showNewPageAndCheckTitleAndButtons(expectedWindowTitle: string, buttons?: ButtonState[]) {
+
+        browser.wait(EC.presenceOf(this.getMainWindow()), BasePage.TIME_OUT_MS_FOR_DIALOG_WINDOW, 'Dialog Window is taking too long to appear in the DOM (timeout: ' + BasePage.TIME_OUT_MS_FOR_DIALOG_WINDOW + ' ms).');
+        this.checkPresentAndDisplayed();
+
+        TestUtil.checkText(this.getTitle(), 'Window title', expectedWindowTitle);
+        if (buttons) {
+            ButtonService.checkButtons(this, buttons);
+        }
+    }
+
+    public searchGridCols(obj: ElementArrayFinder,  strTextToSearch: string): promise.Promise<number>  {
```

Comment:
The method searchGridCols may be at the grid.service.ts

Reply:
done